scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/task/internal/InternalTask.java

```diff
@@ -1229,19 +1230,40 @@ public boolean equals(Object obj) {
      * @param schedulingService used to load results from the database
      */
     public synchronized void updateParentTasksResults(SchedulingService schedulingService) {
-        if (internalTasksDependencies != null && parentTasksResults == null) {
-            Set<TaskId> parentIds = new HashSet<>(internalTasksDependencies.size());
-            for (InternalTask parentTask : internalTasksDependencies) {
-                parentIds.addAll(InternalTaskParentFinder.getInstance().getFirstNotSkippedParentTaskIds(parentTask));
+        if (parentTasksResults == null &&
+            (internalTasksDependencies != null || ifBranch != null || joinedBranches != null)) {
+            Set<TaskId> parentIds = new LinkedHashSet<>();
+            if (internalTasksDependencies != null) {
+                for (InternalTask parentTask : internalTasksDependencies) {
+                    parentIds.addAll(InternalTaskParentFinder.getInstance()
+                                                             .getFirstNotSkippedParentTaskIds(parentTask));
+                }
+            }
+            if (ifBranch != null) {
+                parentIds.add(ifBranch.getId());
+            }
+            if (joinedBranches != null) {
+                for (InternalTask parentTask : joinedBranches) {
+                    parentIds.add(parentTask.getId());
+                }
+            }
+
+            parentTasksResults = new LinkedHashMap<>();
+
+            if (internalTasksDependencies != null) {
```

Comment:
You have the same if twice if (internalTasksDependencies != null)
Maybe you can manage to merge the instructions inside them in a single if block

Reply:
done